pages/resources/tutorials/query-history.md

```diff
@@ -0,0 +1,189 @@
+
+The identifier is a single byte that maps to the fields found in an Ethereum block, for example:
+
+- `0x00`: block header
+- `0x01`: block body
+
+
+This is prepended to the hash of the block that you want to retrieve data from. For example, if you want to retrieve the header from the block with hash `0x6f68209b1294f6365a7ef6c24d41a7195bc68831ae57261b277029196e00f6b0`, your content key would be `0x00` + `6f68209b1294f6365a7ef6c24d41a7195bc68831ae57261b277029196e00f6b0` = `0x006f68209b1294f6365a7ef6c24d41a7195bc68831ae57261b277029196e00f6b0`.
+
+The content key is sent to the node as a parameter is a JSON object along with some metadata about the request. The basic form of a request is as follows:
+
+```json
+{
+    "jsonrpc":"2.0", "method": "<portal-method>", "params":[<content-key>], "id":1
+}
+```
+
+This can be sent to your node over HTTP using curl, as follows:
+
+```sh
+curl -X POST --data '{"jsonrpc":"2.0", "method": "portal_historyRecursiveFindContent", "params":["0x006f68209b1294f6365a7ef6c24d41a7195bc68831ae57261b277029196e00f6b0"], "id":1}'
+```
+
+## Checking available data
+
+Portal Network is being gradually filled up with historical data through bridge nodes. At the current time, not all of Ethereum's historical data has been uploaded into the network. This means that there is a reasonable chance that an arbitrary request will fail because the data is not yet available. The chance of arbitrary requests succeeding will increase as more data is uploaded. The more people run bridge nodes, the faster this process will be.
+
+To check what data has been uploaded to the network, you can use the network health monitoring tool, [Glados](http://glados.ethportal.net/content/).
+
+Glados allows you to view recent uploads into the network on the landing page. However, you can also append `key/<content-key>` to the landing page URL to lookup a specific content key to see if the data is currently available.
+
+Here's an example of a Glados entry for a recent successful upload:
+
+[Key 0x006e0ee92ebaf1b48919d193bd574b3096a85cc2609d286cdc511ad97ce5a18159](http://glados.ethportal.net/content/key/0x006e0ee92ebaf1b48919d193bd574b3096a85cc2609d286cdc511ad97ce5a18159/)
+
+You can tell from then key that this is a block header (begins `0x00` and the block hash was `0x06e0ee92ebaf1b48919d193bd574b3096a85cc2609d286cdc511ad97ce5a18159` )
+
+
+So, to retrieve some real data from your Portal node, you can run the following command (or if you wish, replace the content key with another one that you have verified exists using Glados):
+
+```sh
+curl -X POST --data '{"jsonrpc":"2.0", "method": "portal_historyRecursiveFindContent", "params":["0x006e0ee92ebaf1b48919d193bd574b3096a85cc2609d286cdc511ad97ce5a18159"], "id":1}'
```

Comment:
Tying the tutorial to a specific block hash here is more than fine for the time being. But it might be worth considering that just because a piece of content **was** available, that doesn't guarantee that it will always be available. Thought, we try to build the network with sufficient data redundancy to prevent data from disappearing. Anyways, it could be worth a mention on what to do if the data isn't available from the network. 

Another interesting tutorial angle might be simply to grab the latest block hash from `etherscan` (or maybe a 5-min old block hash to give the network a bit of time to propagate the data) - and then test whether or not that block hash is available in the network.